lib/framebuffer-internal.h

```diff
@@ -86,6 +86,28 @@ class Framebuffer {
       unsigned int output_enable_rev2 : 1;  // 2      (Pi1.Rev2; masks: I2C SDA)
       unsigned int clock_rev2         : 1;  // 3      (Pi1.Rev2; masks: I2C SCL)
 #endif
+#ifdef ADAFRUIT_RGBMATRIX_HAT // assumes 40 pins (A+/B+/Pi2)
```

Comment:
Make this whole ADAFRUIT_RGGMATRIX_HAT around the full type, not selected parts in the type, it makes it hard to read later
So 

```
#ifdef ADAFRUIT_RGBMATRIX_HAT
union IoBits {
   // the HAT needed iobits, starting with
   unsigned int unused _0_3   : 4;  // 0..3
   unsigned int output_enable : 1;  // 4
   // ...
};
#else
union IoBits {
   // the original
}
#endif